robottelo/datafactory.py

```diff
@@ -288,3 +289,124 @@ def valid_usernames_list():
         gen_string("alphanumeric"),
         gen_string("numeric"),
     ]
+
+
+def valid_http_credentials(url_encoded=False):
+    """Returns a list of valid credentials for HTTP authentication
+
+    :param url_encoded: flag for quoting special characters
+    :return: A list of dictionaries with user and password credentials
+    """
+    credentials = [
```

Comment:
Don't want to ping you anymore as it is already too much efforts, but appreciate if you can put any explanation here in the code, like:
'quote' - blah-blah
'http_valid' - blah-blah